test/Microsoft.AspNetCore.Mvc.DataAnnotations.Test/Internal/DataAnnotationsMetadataProviderTest.cs

```diff
@@ -270,6 +275,102 @@ public void CreateDisplayMetadata_DisplayAttribute_DoesNotOverrideDisplayNameAtt
             Assert.Equal("DisplayNameAttributeValue", context.DisplayMetadata.DisplayName());
         }
 
+        [Fact]
+        public void CreateDisplayMetadata_DisplayNameAttribute_OnEnum_CompatSwitchWorks()
+        {
+            // Arrange
+            var unsharedLocalizer = new Mock<IStringLocalizer>(MockBehavior.Strict);
+            unsharedLocalizer
+                .Setup(s => s["DisplayNameValue"])
+                .Returns(new LocalizedString("DisplaynameValue", "didn't use shared"));
+
+            var sharedLocalizer = new Mock<IStringLocalizer>(MockBehavior.Strict);
+            sharedLocalizer
+                .Setup(s => s["DisplayNameValue"])
+                .Returns(
+                () =>
+                {
+                    return new LocalizedString("DisplayNameValue", "used shared");
+                });
+
+            var stringLocalizerFactoryMock = new Mock<IStringLocalizerFactory>(MockBehavior.Strict);
+            stringLocalizerFactoryMock
+                .Setup(s => s.Create(typeof(TestEnum)))
+                .Returns(() => {
+                    return unsharedLocalizer.Object;
+                });
+            stringLocalizerFactoryMock
+                .Setup(s => s.Create(typeof(EmptyClass)))
+                .Returns(() =>
+                {
+                    return sharedLocalizer.Object;
+                });
+
+            var localizationOptions = Options.Create(new MvcDataAnnotationsLocalizationOptions());
+            localizationOptions.Value.AllowDataAnnotationsLocalizationForEnumDisplayAttributes = false;
+            localizationOptions.Value.DataAnnotationLocalizerProvider = (type, stringLocalizerFactory) =>
+            {
+                return stringLocalizerFactory.Create(typeof(EmptyClass));
```

Comment:
Can we throw from here e.g. `throw new Exception("This shouldn't be invoked");`? Presumably, this code path should not be called

Reply:
Actually, it does get invoked because we create the "regular" localizer first, then possibly the enum localizer as opposed to one or the other.